store/postgres/src/fork.rs

```diff
@@ -0,0 +1,347 @@
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex},
+};
+
+use graph::{
+    block_on,
+    components::store::SubgraphFork as SubgraphForkTrait,
+    prelude::{
+        info,
+        r::Value as RValue,
+        reqwest,
+        s::{Definition, Field, ObjectType, TypeDefinition},
+        serde_json, Attribute, DeploymentHash, Entity, Logger, Schema, Serialize, StoreError,
+        Value,
+    },
+    url::Url,
+};
+
+#[derive(Serialize, Debug, PartialEq)]
+struct Query {
+    query: String,
+    variables: Variables,
+}
+
+#[derive(Serialize, Debug, PartialEq)]
+struct Variables {
+    id: String,
+}
+
+/// SubgraphFork represents a simple subgraph forking mechanism
+/// which lazily fetches entities from a remote subgraph's store
+/// associated with a GraphQL endpoint at `fork_url`.
+///
+/// Since this mechanism is used for debug forks, entities are
+/// fetched only once per id in order to avoid fetching an entity
+/// that was deleted from the local store and thus causing inconsistencies.
+pub(crate) struct SubgraphFork {
+    client: reqwest::Client,
+    endpoint: Url,
+    schema: Arc<Schema>,
+    fetched_ids: Mutex<HashSet<String>>,
```

Comment:
The implementation of the fetching mechanism seems correct, @Jannis do you see any problem with using a `Mutex` here? 🤔